tests/test_multidict_.py

```diff
@@ -0,0 +1,203 @@
+            md_multivalue['key99']
+        with pytest.raises(KeyError, match='key99'):
+            md_multivalue.getone('key99')
+
+        assert md_multivalue.getone('key99', 'default') == 'default'
+
+    def test__iter__(self, md_multivalue):
+        assert list(md_multivalue) == ['key1', 'key2', 'key1']
+
+    def test_keys__contains(self, md_multivalue):
+        assert list(md_multivalue.keys()) == ['key1', 'key2', 'key1']
+
+        assert 'key1' in md_multivalue.keys()
+        assert 'key2' in md_multivalue.keys()
+
+        assert 'foo' not in md_multivalue.keys()
+
+    def test_values__contains(self, md_multivalue):
+        assert list(md_multivalue.values()) == ['one', 'two', 3]
+
+        assert 'one' in md_multivalue.values()
+        assert 'two' in md_multivalue.values()
+        assert 3 in md_multivalue.values()
+
+        assert 'foo' not in md_multivalue.values()
+
+    def test_items__contains(self, md_multivalue):
+        assert list(md_multivalue.items()) == [('key1', 'one'), ('key2', 'two'), ('key1', 3)]
+
+        assert ('key1', 'one') in md_multivalue.items()
+        assert ('key2', 'two') in md_multivalue.items()
+        assert ('key1', 3) in md_multivalue.items()
+
+        assert ('foo', 'bar') not in md_multivalue.items()
+
+
+class TestComparisons:
+
+    def test_keys_is_set_less(self, md_simple):
+        assert md_simple.keys() < {'key1', 'key2'}
+
+    def test_keys_is_set_less_equal(self, md_simple):
+        assert md_simple.keys() <= {'key1'}
+
+    def test_keys_is_set_equal(self, md_simple):
+        assert md_simple.keys() == {'key1'}
+
+    def test_keys_is_set_greater(self, md_simple):
+        assert {'key1', 'key2'} > md_simple.keys()
+
+    def test_keys_is_set_greater_equal(self, md_simple):
+        assert {'key1'} >= md_simple.keys()
+
+    def test_keys_is_set_not_equal(self, md_simple):
+        assert md_simple.keys() != {'key2'}
+
+    def test_eq(self, md_simple):
+        assert {'key1': 'value1'} == md_simple
+
+    @pytest.mark.parametrize("cls", MUTABLE_DICTS)
+    def test_eq2(self, cls, md_simple):
+        another_md = cls([('key2', 'value1')])
+        assert md_simple != another_md
+
+    @pytest.mark.parametrize("cls", MUTABLE_DICTS)
+    def test_eq3(self, cls, md_simple):
+        empty_md = cls()
+        assert md_simple != empty_md
+
+    @pytest.mark.parametrize("cls", MUTABLE_DICTS)
+    def test_eq_other_mapping_contains_more_keys(self, cls):
+        d1 = cls(foo='bar')
+        d2 = dict(foo='bar', bar='baz')
+        assert d1 != d2
+
+    def test_ne(self, md_simple):
+        assert md_simple != {'key1': 'another_value'}
+
+    def test_and(self, md_simple):
+        assert {'key1'} == md_simple.keys() & {'key1', 'key2'}
+
+    def test_and2(self, md_simple):
+        assert {'key1'} == {'key1', 'key2'} & md_simple.keys()
+
+    def test_or(self, md_simple):
+        assert {'key1', 'key2'} == md_simple.keys() | {'key2'}
+
+    def test_or2(self, md_simple):
```

Comment:
All test assertions with a reversed order, but checking basically the same should go to the same test.